src/services/ASTM54BService.ts

```
@@ -0,0 +1,276 @@
+    // Calculate temperature difference from reference (15°C)
+    const dT = observedTemp - this.REFERENCE_TEMPERATURE;
+
+    // Calculate thermal expansion coefficient
+    const alpha = this.calculateAlpha(density15c);
+
+    // Apply ASTM 54B formula: VCF = e^(-ALPHA * dT * (1 + 0.8 * ALPHA * dT))
+    const exponent = -alpha * dT * (1 + 0.8 * alpha * dT);
+    const vcf = Math.exp(exponent);
+
+    // Validate result
+    if (!isFinite(vcf) || vcf <= 0 || vcf > 2) {
```

Biome (1.9.4)

[error] isFinite is unsafe. It attempts a type coercion. Use Number.isFinite instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isFinite instead.


(lint/suspicious/noGlobalIsFinite)